warehouse/query-core/src/main/java/datawave/core/iterators/key/util/TFKeyUtil.java

```diff
@@ -0,0 +1,124 @@
+package datawave.core.iterators.key.util;
+
+import org.apache.accumulo.core.data.ByteSequence;
+import org.apache.accumulo.core.data.Key;
+
+/**
+ * A utility for determining key type and accessing the field, value, datatype and uid components
+ * <p>
+ * Note: Each method call will return a new object. Values are not cached, no state is saved.
+ * <p>
+ * A TermFrequency key structure is row:tf:datatype\0uid\0value\0FIELD
+ */
+public class TFKeyUtil {
+    
+    private TFKeyUtil() {
+        // static utility
+    }
+    
+    /**
+     * Determines if the provided key is an instance of a TermFrequency key
+     *
+     * @param k
+     *            the key
+     * @return true if this is a TermFrequency key
+     */
+    public static boolean instanceOf(Key k) {
+        ByteSequence bytes = k.getColumnFamilyData();
+        return bytes != null && bytes.length() == 2 && bytes.byteAt(0) == 't' && bytes.byteAt(1) == 'f';
+    }
+    
+    /**
+     * Parses the field as a String
+     * 
+     * @param k
+     *            the key
+     * @return the field
+     */
+    public static String getFieldString(Key k) {
+        ByteSequence bytes = k.getColumnQualifierData();
+        for (int i = bytes.length() - 1; i >= 0; i--) {
+            if (bytes.byteAt(i) == 0x00) {
+                return bytes.subSequence(i + 1, bytes.length()).toString();
```

Comment:
We have some funny error cases in this class that likely existed in the class you based this on, so it might be OK to leave them. That said it is worth considering whether something different is needed.

For example, if for some reason the offsets when we call `subSequence` are wrong, we'll throw some sort of exception likely, but if we just can't find anything we return `null`. Should we consider treating all failure cases the same?  Not certain whether returning null or throwing an exception is more ideal.